APE24.rst

```diff
@@ -0,0 +1,477 @@
+the pyOpenSci form above, specifying:
+
+* astronomy domain expertise, and
+* membership in the Astropy community.
+
+pyOpenSci does not publicly list all the people who signed up
+(i.e, a reviewer is only public during the review process)
+but the `pyOpenSci Editorial Board`_ has access to the list.
+
+pyOpenSci will respect the Astropy-specific criteria for someone to be a
+reviewer for Affiliated request, namely:
+
+* familiarity with the Astropy project,
+* ability to judge whether a package integrates well with the Astropy ecosystem
+  (as per pre-APE 24 guidelines), and
+* having domain expertise in the area of the package (e.g., galaxy evolution).
+
+pyOpenSci and Astropy both ask editors/reviewers disclose any potential conflict of interest (COI)
+prior to agreeing to review a package. In the event where COI occurs, with this partnership,
+we would follow the
+`pyOpenSci COI process <https://www.pyopensci.org/software-peer-review/our-process/policies.html#conflict-of-interest-for-reviews-and-editors>`_.
+pyOpenSci invites the Astropy community to review this language to ensure it meets our needs.
+
+.. _reviewer sign-up form: https://forms.gle/GHfxvmS47nQFDcBM6
+
+Review turnaround time
+----------------------
+
+Astropy currently does not enforce any concrete turnaround time mainly because
+we do not know who ends up reviewing a given submission nor have control
+over their schedule. Furthermore, we only have two editors who may not have
+sufficient time themselves to actively push a review forward. Therefore, a review
+could take weeks, months, or (gasp) years.
+
+If we partner with pyOpenSci, Astropy would need to adhere to their expected
+timeline for each step, as laid out in `An Overview Of the Peer Review Process`_.
+For example, editor is expected to find reviewers within 2-3 weeks and
+a peer review should be completed within 3 weeks after that.
+This is to ensure that the package maintainer have a good experience with the review
+and things do not languish over a long period of time.
+
+Generally, the editor role should not take a huge amount of time, *but* it is
+important for an editor, once the review starts, to check in on the review periodically
+(every few weeks and more often during wrap-up).
+
+.. _An Overview Of the Peer Review Process: https://www.pyopensci.org/software-peer-review/our-process/review-timeline.html
+
+
+Packaging guidelines
+--------------------
+
+pyOpenSci has been developing a `community-driven packaging guide`_
+that covers modern best practices and recommendations for scientific Python packaging.
+This guide has a stringent community review process. Reviewers for this guide
+represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).
+
+Meanwhile, the Astropy community prefers a more laid-back
+`OpenAstronomy packaging guide <https://packaging-guide.openastronomy.org/en/latest/>`_.
```

Comment:
I am going to object to the "laid-back" wording here, or at least ask for some more specifics.

Also, the OpenAstronomy guide is much narrower in scope to the pyOpenSci one, we describe how to make a package *which looks like astropy*. Now I know this might not be the "best" or "most popular" way to build a package, but it's what we do and what most of our coordinated packages do.

I therefore am less keen on "encouraging our Affilliated packages to follow [the pyOpenSci]" guide unless we are going to do the work of making changes to all our packages as well. (specifically here I am referring to src/ layout, but a few other things as well). Given the amount of work this would be for the debatable returns I can't see it ever happening.

Reply:
@lwasser , do you wish to respond to this? 🙏

Reply:
> I therefore am less keen on "encouraging our Affilliated packages to follow [the pyOpenSci]" guide unless we are going to do the work of making changes to all our packages as well. (specifically here I am referring to src/ layout,

Yes, this has met considerable resistance in astropy/astropy#13678 and I'd rather not reopen that lightly again.

Reply:
i hear you. and am with you 100% . we went through the same process. matplotlib's src layout is actually for (i think) C / compiled code if i recall correctly so it's technically not even a "traditional" src/ layout. there are some packages out there using src/ now BUT people will do what they wish to do here. and it's not in any way a roadblock for this collaboration or for maintainers in general!  we just make recommendations and have accepted packages with both flat and src layouts. sounds like we are all on the same page here! 

for more specific reference [this is our criteria](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements) that our editor in chief looks for upon initial submission. because this text:

**Issue Submission Documentation** 

confuses people (i need to clarify that text!) it refers to the yaml in our review issue being filled out completed nothing to do with the package itself :) 
when we initially compared our checks to astropy's they seemed fairly well aligned. 
i really appreciate this conversation!

Reply:
> i hear you. and am with you 100% . we went through the same process. matplotlib's src layout is actually for (i think) C / compiled code if i recall correctly so it's technically not even a "traditional" src/ layout.

It's using `src` for to-be-compiled code and `lib` for the Python source, but the effect is the same, the package not being directly visible at the top-level directory.
> 
> for more specific reference [this is our criteria](https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements) that our editor in chief looks for upon initial submission. because this text:

That seemed to have some potential for confusion to me as well, as those are just the entry-point criteria, but we may want to also indicate what criteria the referees (and finally the editor again) should use for accepting a package.

Reply:
hey @dhomeier the best place to find all of this information is our peer review guide.

Yes, the Editor in Chief checks serve as a base level of what a package needs to begin review. These checks are similar to those in the astropy review process. We will ask an author to add documentation, etc if that doesn't exist prior to a review beginning. 

* here is the [editor guide](https://www.pyopensci.org/software-peer-review/how-to/editors-guide.html)
* and here is the [reviewer guide](https://www.pyopensci.org/software-peer-review/how-to/reviewer-guide.html#peer-review-template) that contains the template information that reviewers look for and sign off on. 

There is a guide in that chapter for all roles involved of our review process so i suggest that y'all have a look at all of the guides - maintainer / author, editor, editor in chief and reviewer to understand the big picture of our process. 

please let me know if you have any questions.

Reply:
On the pyOpenSci side, is it possible to have the Astropy check override the packaging guideline requirements? After all, we did agree on adding some extra domain-specific stuff.

So, I can see https://www.pyopensci.org/software-peer-review/how-to/author-guide.html#does-your-package-meet-packaging-requirements have slightly qualifiers for different domains. But if someone submit something that is cross-domain, then I think it is reasonable to require pyOpenSci standards over Astropy's.

Reply:
2023-11-10 -- I accepted Leah's edits about packaging and then some more. Hopefully acceptable by everyone here?

Reply:
i will read through this doc in it's entirety one day this week but as far as i'm concerned astropy doesn't have to follow our packaging guidelines. They are recommendations. Things like src/ vs flat layout won't make or break the functionality of usability of a package as far as i'm concerned especially if there are tests and people know to run tests on the installed version of the package. I think src/ will generally make it a bit easier to have a solid test setup for beginners. but that is my opinion.  
We would never reject a package given it has a particular file layout. 

I do have a question for all of you here: is it the case that if someone came to you / us that is astropy related (so a good fit) but was using a src/ layout - would you reject the package or ask them to move back to a flat layout?

i just personally think that - that particularly issue is not an issue we'd ever want to reject a package over. Nor would we want someone who is following the pypa tutorials or setuptools docs and using src/ layout to be forced back to a flat layout. that is my only concern. 

i don't want to debate each layout. i just want to understand the principles here in a review associated with following openastronomy guidelines.

Reply:
ok reading the comments below - https://github.com/astropy/astropy-APEs/pull/87#discussion_r1368832543 it seems like you all are on board with recommending the openastronomy guidelines but wouldn't reject a package because of a specific packaging choice that was totally valid. if that is the case let's resolve this as well.

Reply:
note: we have no requirements related to the layout of the package. I think you can look at our editor in chief checks below and potentially link to it. We might suggest changes if a package has a hard-to-maintain infrastructure but it would never be rejected because of a specific layout (ie src vs flat or using a specific tool - hatch vs pdm or backend etc)

https://www.pyopensci.org/software-peer-review/how-to/editor-in-chief-guide.html#editor-checklist-template